### PR TITLE
Fix window positioning issue on macOS

### PR DESCRIPTION
**Pull request by mteit**
PR addresses an issue related to window positioning on macOS. The problem arises when setting the window to visible after resizing, which leads to inconsistent window positioning behavior after resizing. Specifically, the configuration window's position becomes dependent on the window size, causing it to appear at the bottom left corner of the window instead of the expected top left corner.

To fix this issue, I adjusted the order of operations to set the window visible before resizing.

This PR is linked to the corresponding GitHub issue: [Link to GitHub issue](https://github.com/alacritty/alacritty/issues/7774)

### CODE REVIEW

File: alacritty/src/display/mod.rs [hunk 1]
```diff
@@ -412,6 +412,9 @@ impl Display {
         let (cell_width, cell_height) = compute_cell_size(config, &metrics);
 
         // Resize the window to account for the user configured size.
+        // In macOS, for keeping the window position after resizing
+        // at the top left window corner, it must be visible.
+        window.set_visible(true);
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:417]: Did you test this on other platforms? The current location of this call is where it is for a reason iirc.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/display/mod.rs (hunk 1 above)
Comment (mteit) [reply alacritty/src/display/mod.rs]: Unfortunately, I wasn't able to test the changes on other platforms. However, I've made adjustments to ensure platform-specific behavior. The call to set the window visible is now conditional based on the target platform, so it should maintain the expected behavior on macOS while not affecting other platforms. Is this approach acceptable to you?

**Comment by chrisduerr** [comment]
>  marked this pull request as draft

Why did you switch the PR to draft status? Did you encounter any issues?

**Comment by mteit** [comment]
> Why did you switch the PR to draft status? Did you encounter any issues?
@chrisduerr I encountered several unexpected behaviors while testing the PR.
Initially, the fix addressed the issue with window position along the `y` axis when `window.decorations = "None"`, ensuring that the position adhered to the configuration in other cases.
However, the commit unexpectedly resolved an issue where `window.dimensions` was restricted in width when `window.decorations = "Full"|"Transparent"|"Buttonless"`.
Despite this fix, I discovered a new issue: when `window.decorations = "Full"|"Transparent"|"Buttonless"`, the window position along the `x` axis starts at 346 pixels (on my display), deviating from the expected behavior. Importantly, this issue persists both on my branch and on the master branch.

So, I'm going to fix the `x` axis issue, and then I will switch the PR status.

File: alacritty/src/display/mod.rs [hunk 2]
```diff
@@ -411,12 +413,23 @@ impl Display {
...
         if let Some(dimensions) = config.window.dimensions() {
             let size = window_size(config, dimensions, cell_width, cell_height, scale_factor);
             window.request_inner_size(size);
         }
 
+        // Set the window position to account for the user configured position.
+        #[cfg(target_os = "macos")]
+        if let Some(position) = config.window.position {
+            window.set_outer_position(PhysicalPosition::new(position.x as u32, position.y as u32));
+        }
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:431]: Did you just add this using trial and error? We're already setting the position on startup, why are you setting it again?

This does not seem like the proper solution, it just looks like we're moving it a bunch hoping that one of the operations will stick.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

**Comment by chrisduerr** [comment]
PR is clearly dead. If someone wants to pick it up, please open a new PR.

**Pull request closed without merging**